### PR TITLE
Add LLD Linker Improvements Project

### PR DESCRIPTION
**Pull request by Prabhuk**
In our efforts to support embedded projects, we noticed that improvements to LLD can help in easing the transition to Clang toolchain. Listing some of those identified items under this project proposal.

### CODE REVIEW

File: OpenProjects.html [hunk 1]
```diff
@@ -27,6 +27,7 @@
               <li><a href="#llvm_module_inliner">Evaluate and Expand the Module-Level Inliner</a></li>
               <li><a href="#llvm_undef_load">Remove undef: move uninitialized memory to poison</a></li>
               <li><a href="#llvm_abi_export">Add ABI/API export annotations to the LLVM build</a></li>
+              <li><a href="#llvm_lld_embedded">LLD Linker Improvements for Embedded Targets</a></li>
```
Comment (asl) [inline OpenProjects.html:30]: why have you added to 2022 section?

File: OpenProjects.html [hunk 2]
```diff
@@ -775,6 +776,56 @@
...
+        mapping between the symbol and the section groups, which is useful for debugging.
+        Preserving this information during the linking process will require modifications to
+        internal linker data structures.</p>
+    </li>
+  </ul>
+  <p><b>Project size:</b> 240 hr</p>
+  <p><b>Difficulty:</b> Medium/Hard</p>
+  <p><b>Skills:</b> C++</p>
+  <p><b>Mentors:</b>
+    <a href="https://www.linkedin.com/in/prabhu-karthikeyan-rajasekaran-2a794412">Prabhu Rajasekaran</a>
```
Comment (asl) [inline OpenProjects.html:823]: I don't think linkedin is good channel for communications with students :)

There is no discourse link as well. There are no expected results.

**Review by asl — CHANGES_REQUESTED** [review]
See comments above

File: OpenProjects.html (hunk 1 above)
Comment (Prabhuk) [reply OpenProjects.html]: It was an oversight on my part. Moved the title to 2023 section.

File: OpenProjects.html (hunk 2 above)
Comment (Prabhuk) [reply OpenProjects.html]: Added Expected Outcomes section. Replaced the linkedin links with github links.

**Comment by asl** [comment]
@Prabhuk Will you please rebase you branch, so it could be applied clearly with linear history?

**Comment by Prabhuk** [comment]
Apologies on the rebase mess. Thanks for moving the discourse topic as well. Please take a look.

**Pull request merged**